### PR TITLE
preProcess and preResponse

### PR DESCRIPTION
**Pull request by dawnerd**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -27,6 +27,13 @@ exports.register = function(server, options, next) {
   const renderHandler = function(viewConfig) {
     return function(request, reply) {
       async.autoInject({
+        preProcess: done => {
+          if (typeof options.preProcess === 'string') {
```
Comment (jgallen23) [inline index.js]: dont need to check if its a string, str2fn will call it if it's a function - like if somebody was using outside hapi-confi

Comment (jgallen23) [inline index.js]: probably just want to check if it exists

File: index.js [hunk 2]
```diff
@@ -61,6 +68,13 @@ exports.register = function(server, options, next) {
         if (request.query.json === '1') {
           return reply(combinedData).type('application/json');
         }
+
+        if (typeof options.preResponse === 'string') {
+          return str2fn(server.methods, options.preResponse)(request, options, combinedData, reply, () => {
+            reply.view(viewConfig.view, combinedData);
```
Comment (jgallen23) [inline index.js]: should it do some sort of check if it should render the view? How do we have it redirect?

File: index.js (hunk 1 above)
Comment (jgallen23) [inline index.js]: should we add way for it to stop early and not process everything?

File: index.js (hunk 2 above)
Comment (dawnerd) [inline index.js]: If you need to override you just call reply in your method and ignore the next param so the reply.view isn't called.

File: index.js (hunk 1 above)
Comment (dawnerd) [inline index.js]: Oh good catch, meant to add reply into the args.

File: index.js (hunk 2 above)
Comment (jgallen23) [inline index.js]: got it

**Pull request merged**